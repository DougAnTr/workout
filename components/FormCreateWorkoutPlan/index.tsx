import { KeyboardAvoidingView, Text } from 'react-native';
import { Input } from '../Input';
import { Divider } from '../Divider';
import { Button } from '../Button';

export function FormCreateWorkoutPlan() {
  return (
    <KeyboardAvoidingView>
      <Text>Create your workout plan</Text>

      <Input placeholder={'Plan name'} />
      <Input placeholder={'Description'} />

      <Divider />

      <Text>Workouts</Text>

      <Button>Add workout</Button>
    </KeyboardAvoidingView>
  );
}
