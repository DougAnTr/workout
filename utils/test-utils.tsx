import { render, RenderOptions } from '@testing-library/react-native'
import { PropsWithChildren, ReactElement } from 'react'

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return (
    <>
      {children}
    </>
  )
}

const customRender = (ui: ReactElement<any>, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react-native'

// override render method
export { customRender as render }
