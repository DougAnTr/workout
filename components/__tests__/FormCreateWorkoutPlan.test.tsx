import { render } from '../../utils/test-utils';
import { FormCreateWorkoutPlan } from '../FormCreateWorkoutPlan';

describe('<FormCreateWorkoutPlan/>', () => {
  it('should render correctly', () => {
    const snapshot = render(<FormCreateWorkoutPlan />).toJSON();

    expect(snapshot).toMatchSnapshot();
  })
})
