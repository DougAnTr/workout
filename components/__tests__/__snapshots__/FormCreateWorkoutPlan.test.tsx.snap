// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormCreateWorkoutPlan/> should render correctly 1`] = `
<View
  onLayout={[Function]}
>
  <Text>
    Create your workout plan
  </Text>
  <TextInput
    placeholder="Plan name"
    testID="input"
  />
  <TextInput
    placeholder="Description"
    testID="input"
  />
  <View
    style={
      {
        "backgroundColor": "gray",
        "height": 1,
        "width": "100%",
      }
    }
  />
  <Text>
    Workouts
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Add workout
    </Text>
  </View>
</View>
`;
